catalog/registry/catalog-registry-federationadminservice/src/main/java/ddf/catalog/registry/federationadmin/service/FederationAdminService.java

```diff
@@ -27,29 +25,136 @@
+     * @return Id of the created metacard
+     * @throws FederationAdminException If the metacard provided was null.
+     *                                  The CatalogFramework.create call throws IngestException or SourceUnavailableException
+     */
+    String addRegistryEntry(Metacard metacard) throws FederationAdminException;
+
+    /**
+     * Add the given metacard to the registry catalog with the provided destinations
+     *
+     * @param metacard     Metacard to be stored
+     * @param destinations Set of destinations to add
+     * @return Id of the created metacard
+     * @throws FederationAdminException If the metacard provided was null.
+     *                                  The CatalogFramework.create call throws IngestException or SourceUnavailableException
+     */
+    String addRegistryEntry(Metacard metacard, Set<String> destinations)
+            throws FederationAdminException;
+
+    //    /**
+    //     * Transforms the {@link RegistryPackageType} to a {@link Metacard} and
+    //     * adds the metacard the registry.
+    //     *
+    //     * @param registryPackage Registry Package to be stored
+    //     * @throws FederationAdminException If the Registry Packge provided is null. Passes exception thrown by addRegistryEntry(Metacard).
+    //     */
+    //    void addRegistryEntry(RegistryPackageType registryPackage) throws FederationAdminException;
+
+    /**
+     * * Deletes the registry entry from the registry catalog for each of the provided registry ids.
+     *
+     * @param registryIds The list of registry ids to be deleted.
+     * @throws FederationAdminException If the list of ids provided is empty.
+     *                                  If IngestException or Source UnavailableException was thrown by CatalogFramework.delete
+     */
+    void deleteRegistryEntriesByRegistryIds(List<String> registryIds)
+            throws FederationAdminException;
+
+    /**
+     * * Deletes the registry entry from the registry catalog for each of the provided metacard ids.
+     *
+     * @param metacardIds The list of metacard ids to be deleted.
+     * @throws FederationAdminException If the list of ids provided is empty.
+     *                                  If IngestException or Source UnavailableException was thrown by CatalogFramework.delete
+     */
+    void deleteRegistryEntriesByMetacardIds(List<String> metacardIds)
+            throws FederationAdminException;
+
+    /**
+     * * Write the provided metacard to the registry catalog. Updating the metacard currently stored.
+     *
+     * @param metacard Metacard with updates to be stored in the registry catalog
+     * @throws FederationAdminException If the provided metacard doesn't have an id.
+     *                                  If CatalogFramework.update call throws IngestException or SourceUnavailableException
+     */
+    void updateRegistryEntry(Metacard metacard) throws FederationAdminException;
```

Comment:
I think we want to be able give destinations and work with an xml string when doing updates like we support for the addRegistryEntries